include/ginkgo/core/matrix/dense.hpp

```diff
@@ -903,10 +922,12 @@ class Dense
      */
     template <typename ValuesArray>
     Dense(std::shared_ptr<const Executor> exec, const dim<2>& size,
-          ValuesArray&& values, size_type stride)
+          ValuesArray&& values, size_type stride,
+          const strategy_type strategy = strategy_type::gko)
         : EnableLinOp<Dense>(exec, size),
           values_{exec, std::forward<ValuesArray>(values)},
-          stride_{stride}
+          stride_{stride},
+          strategy_(strategy)
```

Comment:
If we accept my suggestion on multiple strategies, we could copy this argument to all strategies, and provide another constructor that takes all the individual strategies.